test/e2e/diagnostics/testYamlWithoutEE.test.ts

```diff
@@ -25,7 +26,96 @@ export function testDiagnosticsYAMLWithoutEE(): void {
 
       it("should provide diagnostics with YAML validation (with ansible-lint)", async function () {
         await activate(docUri1);
+        console.log(
+          "Extension activated, waiting for language server to be ready...",
+        );
+
+        // Wait for language server to be ready by checking for completion providers
+        let serverReady = false;
+        let attempts = 0;
+        const maxAttempts = 20;
+
+        while (!serverReady && attempts < maxAttempts) {
+          try {
+            // Try to get hover information as a test of server readiness
+            const hovers = await vscode.commands.executeCommand(
+              "vscode.executeHoverProvider",
+              docUri1,
+              new vscode.Position(0, 0),
+            );
+            if (hovers) {
+              serverReady = true;
+              console.log(
+                `Language server ready after ${attempts + 1} attempts`,
+              );
+            }
+          } catch {
+            // Server not ready yet, will retry
+            console.log(
+              `Attempt ${attempts + 1}: Language server not ready yet`,
+            );
+          }
+
+          if (!serverReady) {
+            await sleep(500);
+            attempts++;
+          }
+        }
+
+        if (!serverReady) {
+          console.log(
+            "Language server may not be fully ready, continuing anyway...",
+          );
+        }
+
+        console.log("Waiting additional time for full initialization...");
+        await sleep(2000); // Give more time for the extension to be ready
```

Comment:
```suggestion
        if (!serverReady) {
          await sleep(2000); // Give more time for the extension to be ready
          console.log("Waiting additional time for full initialization...");

          console.log(
            "Language server may not be fully ready, continuing anyway...",
          );
        }
```
Let's only do this final sleep if the server still isn't ready, so we're not waiting again if it is ready